src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/BannedAssembliesAnalyzer.cs

```diff
@@ -86,5 +89,15 @@ static void CheckType(ISymbolAnalysisContext context, ITypeSymbol type, ISymbol
                     break;
             }
         }
+
+        private static bool IsUtf8JsonReaderWriter(ITypeSymbol type)
+        {
+            return (type.Name == "Utf8JsonReader" || type.Name == "Utf8JsonWriter") && GetFullNamespace(type.ContainingNamespace) == "System.Text.Json";
+        }
+
+        private static string GetFullNamespace(INamespaceSymbol namespaceSymbol)
+        {
+            return namespaceSymbol.ContainingNamespace.Name == "" ? namespaceSymbol.Name : $"{GetFullNamespace(namespaceSymbol.ContainingNamespace)}.{namespaceSymbol.Name}";
```

Comment:
what is this line supposed to do? especially "{GetFullNamespace(namespaceSymbol.ContainingNamespace)}.{namespaceSymbol.Name}"? it seems to concatenating two namespace names.

Reply:
Yes, `namespaceSymbol.Name` only contains current name of current namespace. For example, for `System.Text.Json.Utf8JsonReader`, its containing namespace is `Json` whose containing namespace is `Text` and so on. Therefore, I have to have a recursive way to get the full namespace.